src/TransactionsBundle/Service/MatchService.php

```diff
@@ -28,37 +28,51 @@ public function __construct(
         $this->entityManager = $entityManager;
     }
 
-    public function match($matches, $transaction, $category)
+    public function match($matches, $openTransaction, $category)
     {
-        $matchDescription = $this->cleanUp($transaction->getDescription());
-        $type = $this->categoryRepository->findById($category);
-
-        if ($transaction->getCategories() != null) {
+        if ($openTransaction->getCategories() != null) {
```

Comment:
Can maybe be called prior to calling `match()`, so you'd move the if statement out of `match()`